presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoDriverUri.java

```diff
@@ -103,10 +109,14 @@ private PrestoDriverUri(URI uri, Properties driverProperties)
     {
         this.uri = requireNonNull(uri, "uri is null");
         address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
-        properties = mergeConnectionProperties(uri, driverProperties);
+        properties = mergeProvidedProperties(uri, driverProperties);
+
+        setSSLPropertyByPortIfNotProvided(properties, uri);
 
         validateConnectionProperties(properties);
 
+        setDefaultPropertiesIfNotProvided(properties);
```

Comment:
If we set default values to `SSL` and `SSL_VERIFICATION`, in the following case, the previous code will throw an exception `java.sql.SQLException: Connection property 'SSLVerification' is not allowed`:

- Both of `SSL` and `SSL_VERIFICATION` are not provided in the URI or driver properties
- The default values of `SSL` and `SSL_VERIFICATION` are merged into the variable `properties`
- `SSL` is default `false`
- `SSL=false` and `SSL_VERIFICATION` in `properties` => Exception

I tried to solve this problem by first validating the provided properties, then merge the default properties into the variable `properties` without overwriting.